src/WebJobs.Extensions.DurableTask/Listener/OutOfProcOrchestrationShim.cs

```diff
@@ -104,6 +116,149 @@ internal async Task<bool> ScheduleDurableTaskEvents(OrchestrationInvocationResul
             return true;
         }
 
+        /// <summary>
+        /// Invokes a DF API based on the input action object.
+        /// Most APIs will return a Task to be awaited, but others such as SignalEntity do not.
+        /// </summary>
+        /// <param name="action">An OOProc action object representing a DF task.</param>
+        /// <param name="returnfireAndForgetTask">Whether to return a `CompletedTask` to represent fire-and-forget APIs. Defaults to `true`.</param>
+        /// <returns>If the API returns a task, the DF task corresponding to the input action. Else, null.</returns>
+        private Task InvokeAPIFromAction(AsyncAction action, bool returnfireAndForgetTask = true)
+        {
+            // Helper function to obtain a list of tasks for compoundTasks.
+            Func<AsyncAction[], Task[]> mapInvokeAPIFromAction = (actions) =>
```

Comment:
Consider moving this out into a static method, or even just replace it with a simple `.Select(this.InvokeApiFromAction)` linq query. The current approach creates `mapInvokeApiFromAction` per method execution, and considering this is a recursive function, it may be called quite often.

Reply:
Sweet! I'm always glad to use LINQ and write neat functional code :)